clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractMembershipManager.java

```diff
@@ -622,10 +631,10 @@ public void transitionToSendingLeaveGroup(boolean dueToExpiredPollTimer) {
 
     /**
      * Call all listeners that are registered to get notified when the member epoch is updated.
-     * This also includes the latest member ID in the notification. If the member fails or leaves
-     * the group, this will be invoked with empty epoch and member ID.
+     * This also includes the member ID in the notification. If the member fails or leaves
+     * the group, this will be invoked with empty epoch.
      */
-    void notifyEpochChange(Optional<Integer> epoch, Optional<String> memberId) {
+    void notifyEpochChange(Optional<Integer> epoch, String memberId) {
```

Comment:
`memberId` is immutable now so we can remove `memberId` from `notifyEpochChange`